client/dashboard/task-list/tasks/payments/paypal.js

```diff
@@ -2,16 +2,255 @@
+import apiFetch from '@wordpress/api-fetch';
+import { Button, TextControl } from 'newspack-components';
+import { getQuery } from '@woocommerce/navigation';
+import { withDispatch } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
+import interpolateComponents from 'interpolate-components';
 
 /**
  * WooCommerce dependencies
  */
+import { WC_ADMIN_NAMESPACE } from 'wc-api/constants';
+import { Form, Link } from '@woocommerce/components';
+import withSelect from 'wc-api/with-select';
 
 class PayPal extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			connectURL: '',
+			showManualConfiguration: props.manualConfig,
+		};
+
+		this.updateSettings = this.updateSettings.bind( this );
+	}
+
+	componentDidMount() {
+		const { showManualConfiguration } = this.state;
+
+		const query = getQuery();
+		// Handle redirect back from PayPal
+		if ( query[ 'paypal-connect' ] ) {
+			if ( '1' === query[ 'paypal-connect' ] ) {
+				this.props.markConfigured( 'paypal' );
+				this.props.createNotice(
+					'success',
+					__( 'PayPal connected successfully.', 'woocommerce-admin' )
+				);
+				return;
+			}
+
+			/* eslint-disable react/no-did-mount-set-state */
+			this.setState( {
+				showManualConfiguration: true,
+			} );
+			/* eslint-enable react/no-did-mount-set-state */
+			return;
+		}
+
+		if ( ! showManualConfiguration ) {
+			this.fetchOAuthConnectURL();
+		}
+	}
+
+	componentDidUpdate( prevProps, prevState ) {
+		if (
+			true === prevState.showManualConfiguration &&
+			false === this.state.showManualConfiguration
+		) {
+			this.fetchOAuthConnectURL();
+		}
+
+		if ( false === prevProps.optionsIsRequesting && true === this.props.optionsIsRequesting ) {
+			this.props.setRequestPending( true );
+		}
+
+		if ( true === prevProps.optionsIsRequesting && false === this.props.optionsIsRequesting ) {
+			this.props.setRequestPending( false );
+		}
+	}
+
+	async fetchOAuthConnectURL() {
+		this.props.setRequestPending( true );
+		try {
+			const result = await apiFetch( {
+				path: WC_ADMIN_NAMESPACE + '/onboarding/plugins/connect-paypal',
+				method: 'POST',
+			} );
+			if ( ! result || ! result.connectUrl ) {
+				this.props.setRequestPending( false );
+				this.setState( {
+					showManualConfiguration: true,
+				} );
+				return;
+			}
+			this.props.setRequestPending( false );
+			this.setState( {
+				connectURL: result.connectUrl,
+			} );
+		} catch ( error ) {
+			this.props.setRequestPending( false );
+			this.setState( {
+				showManualConfiguration: true,
+			} );
+		}
+	}
+
+	renderConnectButton() {
+		const { connectURL } = this.state;
+		return (
+			<Button isPrimary isDefault href={ connectURL }>
+				{ __( 'Connect', 'woocommerce-admin' ) }
+			</Button>
+		);
+	}
+
+	async updateSettings( values ) {
+		const { createNotice, isSettingsError, updateOptions, markConfigured } = this.props;
+
+		this.props.setRequestPending( true );
+		await updateOptions( {
```

Comment:
I think this await pattern we've been using will not work with wc-api.  I'm not sure it returns the promise and `isSettingsError` will not yet have updated.

Okay, to handle this in a follow-up as well since we have a few other cases of this that I've also introduced and need correcting.

Reply:
Logged a follow-up issue for us: https://github.com/woocommerce/woocommerce-admin/issues/2959